src/OpenLoco/src/Windows/CompanyList.cpp

```diff
@@ -1154,47 +1116,49 @@ namespace OpenLoco::Ui::Windows::CompanyList
                     continue;
 
                 auto colour = _cargoLineColour[i];
+                auto* deliveredCargoPayment = Economy::getDeliveryCargoPaymentsTable(i);
 
-                _graphSettings->yData[count] = reinterpret_cast<uint32_t*>(&_deliveredCargoPayment[i][0]);
-                _graphSettings->dataStart[count] = 0;
-                _graphSettings->lineColour[count] = Colours::getShade(colour, 6);
-                _graphSettings->itemId[count] = i;
+                _graphSettings.yData[count] = reinterpret_cast<std::byte*>(deliveredCargoPayment);
+                _graphSettings.dataStart[count] = 0;
+                _graphSettings.lineColour[count] = Colours::getShade(colour, 6);
+                _graphSettings.itemId[count] = i;
                 count++;
             }
 
-            _graphSettings->lineCount = count;
-            _graphSettings->dataEnd = 60;
-            _graphSettings->dataTypeSize = 4;
-            _graphSettings->xLabel = StringIds::cargo_delivered_days;
-            _graphSettings->yLabel = StringIds::cargo_delivered_currency;
-            _graphSettings->word_113DD80 = 5;
-            _graphSettings->xAxisLabelIncrement = 20;
-            _graphSettings->dword_113DD86 = 0;
-            _graphSettings->dword_113DD8A = 0;
-            _graphSettings->dword_113DD8E = 0;
+            _graphSettings.lineCount = count;
+            _graphSettings.dataEnd = 60;
```

Comment:
if we used a proper type for getDeliveryCargoPaymentsTable this could then use the `size()` of that proper type